### PR TITLE
Move backend "spvm" to Utils::Backends

### PR DESCRIPTION
**Pull request by Zaoliang**
see https://progress.opensuse.org/issues/33388
verification runs:
https://openqa.suse.de/tests/2915205 (s390x)
https://openqa.suse.de/tests/2915204 (ppc64le)
https://openqa.suse.de/tests/2912003 (aarch64)
https://openqa.suse.de/tests/2915206 (spvm)
https://openqa.suse.de/tests/2915201 (xen)
https://openqa.suse.de/tests/2912213 (svirt-xen-pv)
https://openqa.suse.de/tests/2915202 (s390x-kvm)
https://openqa.suse.de/tests/2915203 (svirt-hyperv)
failed at hostname_inst: it should not be there, compared with
related tests on osd

### CODE REVIEW

File: lib/Utils/Backends.pm [hunk 1]
```diff
@@ -83,4 +84,9 @@ sub is_svirt_except_s390x {
 
 }
 
+sub is_spvm_backend {
+    # PowerVM backend, poo#33388
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Please properly add the documentation, (check #7495).

File: lib/Utils/Backends.pm [hunk 2]
```diff
@@ -83,4 +84,9 @@ sub is_svirt_except_s390x {
 
 }
 
+sub is_spvm_backend {
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Please rename to is_spvm.

Comment (foursixnine) [reply lib/Utils/Backends.pm]: You also will need to update the call to is_remote_backend, accordingly.

File: lib/bootloader_setup.pm [hunk 3]
```diff
@@ -903,18 +903,8 @@ sub zkvm_add_disk {
 
 sub zkvm_add_pty {
     my ($svirt) = shift;
-
```
Comment (foursixnine) [inline lib/bootloader_setup.pm]: If this is about spvm, why is this code being changed? Looks unrelated.

File: lib/Utils/Backends.pm (hunk 1 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: done, thanks

File: lib/main_common.pm [hunk 4]
```diff
@@ -952,7 +952,7 @@ sub load_inst_tests {
             and !get_var("REMOTE_CONTROLLER")
             and !is_hyperv_in_gui
             and !is_bridged_networking
-            and get_var('BACKEND', '') !~ /ipmi|s390x|spvm/
+            and (get_var('BACKEND', '') !~ /ipmi|s390x/) || !is_spvm
```
Comment (foursixnine) [inline lib/main_common.pm]:
```suggestion
            and (get_var('BACKEND', '') !~ /ipmi|s390x/ || !is_spvm)
```

File: lib/Utils/Backends.pm [hunk 5]
```diff
@@ -13,6 +13,16 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
+=head1 Backends.pm
+
+=head1 SYNOPSIS
+
+use Utils::Backends
+
+This is Backends of Utils, it provides various backends to other modules
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Please try to describe better what this library is doing. This message doesn't make much sense.

File: lib/Utils/Backends.pm (hunk 2 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: yes

File: lib/Utils/Backends.pm [hunk 6]
```diff
@@ -13,6 +13,17 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
+=head1 Backends.pm
+
+=head1 SYNOPSIS
+
+use Utils::Backends
+
+It defines various backend functions. These backends in "BACKEND" and "CONSOLES"
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]:
```suggestion
It defines various functions that allows to check for different backend or console types. It exports C<CONSOLES> and C<BACKEND> 
```

File: lib/Utils/Backends.pm [hunk 7]
```diff
@@ -47,40 +59,84 @@ our %EXPORT_TAGS = (
...
 sub use_ssh_serial_console {
     select_console('root-ssh');
     $serialdev = 'sshserial';
     set_var('SERIALDEV', $serialdev);
     bmwqemu::save_vars();
 }
 
+=head2 is_remote_backend
+
+remote backend
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Please try to explain better what the method is doing...

File: lib/Utils/Backends.pm [hunk 8]
```diff
@@ -47,40 +59,84 @@ our %EXPORT_TAGS = (
...
+    return check_var('ARCH', 's390x') || (get_var('BACKEND', '') =~ /ipmi|svirt/) || is_spvm();
 }
 
 # In some cases we are using a VNC connection provided by the hypervisor that
 # allows access to the ttys same as for accessing any remote libvirt instance
 # but not what we use for s390x-kvm.
+
+=head2 has_ttys
+
+remote backend using ttys for: ipmi, s390x, spvm, except S390_ZKVM
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Please combine this with the brief comment that is above this method.

File: lib/Utils/Backends.pm [hunk 9]
```diff
@@ -13,6 +13,15 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
+=head1 Backends.pm
```
Comment (SergioAtSUSE) [inline lib/Utils/Backends.pm]: I think this shouldn't have the file extension

```suggestion
=head1 Backends
```

File: lib/Utils/Backends.pm (hunk 7 above)
Comment (SergioAtSUSE) [reply lib/Utils/Backends.pm]: Also for the others. Something like:

"Returns true if the current instance is running as a X backend"

File: lib/Utils/Backends.pm (hunk 6 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: yes, thanks

File: lib/Utils/Backends.pm (hunk 8 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: updated, thanks

File: lib/Utils/Backends.pm (hunk 9 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: changed

File: lib/Utils/Backends.pm (hunk 7 above)
Comment (Zaoliang) [reply lib/Utils/Backends.pm]: changed

**Comment by okurz** [comment]
minor point I just saw: typo in the commit message "Utilis" :)

File: lib/Utils/Backends.pm [hunk 10]
```diff
@@ -13,6 +13,15 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
+=head1 Backends
+
+=head1 SYNOPSIS
+
+use Utils::Backends
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Somehow you removed the empty line. 
```suggestion
use Utils::Backends

```

File: lib/Utils/Backends.pm [hunk 11]
```diff
@@ -47,40 +57,84 @@ our %EXPORT_TAGS = (
...
+
 sub is_svirt_except_s390x {
     return !get_var('S390_ZKVM') && check_var('BACKEND', 'svirt');
+}
+
+=head2 is_spvm 
+
+Returns true if the current instance is running as PowerVM backend 'spvm'
 
+=cut
```
Comment (foursixnine) [inline lib/Utils/Backends.pm]: Empty line missing.

**Comment by Zaoliang** [comment]
Related verification tests runs on osd re-triggered and they look good.

**Comment by foursixnine** [comment]
> Related verification tests runs on osd re-triggered and they look good.

Link?

**Comment by Zaoliang** [comment]
new test verification:
https://openqa.suse.de/tests/2919044
https://openqa.suse.de/tests/2919045
https://openqa.suse.de/tests/2919046
https://openqa.suse.de/tests/2919047
https://openqa.suse.de/tests/2919050
https://openqa.suse.de/tests/2912213
https://openqa.suse.de/tests/2919053
https://openqa.suse.de/tests/2919056

**Comment by SergioAtSUSE** [comment]
We need to find the reason why these two jobs are failing in your verification runs:
- https://openqa.suse.de/tests/2919044#step/hostname_inst/5
- https://openqa.suse.de/tests/2919047#step/hostname_inst/7

They are not failing in the productive machines: https://openqa.suse.de/tests/overview?arch=&modules=hostname_inst&distri=sle&version=15-SP1&build=228.2&groupid=110#

Anyway, we should merge this after build validation.

**Comment by Zaoliang** [comment]
hostname_inst should not be there as I already mentioned this morning. someone has changed main_common.pm, but this is not relevant anyway.

**Comment by SergioAtSUSE** [comment]
@Zaoliang, please, notice that in 228.2 it is working: https://openqa.suse.de/tests/2921578#step/hostname_inst/8

Could you clone those two scenarios with build 228.2?

**Comment by Zaoliang** [comment]
https://openqa.suse.de/tests/2927101#step/hostname_inst/8

it works fine now with my PR. 
But if you check that the test module failed 19 day ago, but anyway my PR won't break it.

**Comment by mimi1vx** [comment]
https://openqa.suse.de/tests/overview?build=0027&groupid=221&version=15&distri=sle ...
